src/Aspire.Hosting/Orchestrator/ParameterProcessor.cs

```diff
@@ -352,5 +362,71 @@ await notificationService.PublishUpdateAsync(parameter, s =>
                 }
             }
         }
+
+        // Save deployment state at the end if modified
+        if (stateModified && deploymentState is not null)
+        {
+            try
+            {
+                deploymentState["Parameters"] = parametersSection;
+                await deploymentStateManager.SaveStateAsync(deploymentState).ConfigureAwait(false);
+                logger.LogInformation("Parameter values saved to deployment state.");
+            }
+            catch (Exception ex)
+            {
+                logger.LogWarning(ex, "Failed to save parameter values to deployment state.");
+            }
+        }
+    }
+
+    private async Task SaveParametersToDeploymentStateAsync(IEnumerable<ParameterResource> parameters, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var deploymentState = await deploymentStateManager.LoadStateAsync(cancellationToken).ConfigureAwait(false);
+            var parametersSection = new JsonObject();
+
+            foreach (var parameter in parameters)
+            {
+                // Try to get the value - skip parameters that don't have values yet
```

Comment:
This logic doesn't seem right. All values should be resolved by the time we get here. Revert back to the simplified logic of getting parameter.ValueInternal